hydra-node/src/Hydra/HeadLogic.hs

```diff
@@ -785,17 +783,24 @@ update Environment{party, signingKey, otherParties} ledger st ev = case (st, ev)
     ) ->
       onOpenNetworkAckSn parties otherParty parameters previousRecoverableState snapshotSignature headState sn
   ( prev@OpenState{parameters, coordinatedHeadState}
-    , OnChainEvent (Observation OnCloseTx{snapshotNumber = closedSnapshotNumber, remainingContestationPeriod})
+    , OnChainEvent (Observation OnCloseTx{snapshotNumber = closedSnapshotNumber, contestationDeadline})
     ) ->
-      onOpenChainCloseTx parameters prev coordinatedHeadState closedSnapshotNumber remainingContestationPeriod
+      onOpenChainCloseTx parameters prev coordinatedHeadState closedSnapshotNumber contestationDeadline
   (ClosedState{confirmedSnapshot}, OnChainEvent (Observation OnContestTx{snapshotNumber})) ->
     onClosedChainContestTx confirmedSnapshot snapshotNumber
-  (ClosedState{}, ShouldPostFanout) ->
-    OnlyEffects [ClientEffect ReadyToFanout]
+  (cst@ClosedState{contestationDeadline, readyToFanoutSent}, OnChainEvent (Tick chainTime))
+    | chainTime > contestationDeadline && not readyToFanoutSent ->
+      NewState
+        -- XXX: Requires -Wno-incomplete-record-updates. Should refactor
+        -- 'HeadState' to hold individual 'ClosedState' etc. types
+        (cst{readyToFanoutSent = True})
```

Comment:
Or simply write: `ClosedState{contestationDeadline, readyToFanoutSent = True}` :sweat_smile: ?

Reply:
Unfortunately it's some more fields 
```
  | ClosedState
      { parameters :: HeadParameters
      , confirmedSnapshot :: ConfirmedSnapshot tx
      , previousRecoverableState :: HeadState tx
      , contestationDeadline :: UTCTime
      , -- | Tracks whether we have informed clients already about being
        -- 'ReadyToFanout'.
        readyToFanoutSent :: Bool
      }
```
and I chose to disable the warning instead of the field noise